webform.tokens.inc

```diff
@@ -297,3 +277,37 @@ function webform_tokens($type, $tokens, array $data = array(), array $options =
 
   return $replacements;
 }
+
+/**
+ * Provides default webform token modifier behavior.
+ *
+ * @param string $modifier
+ *   The modifier that is being used.
+ * @param string $display_element
+ *   The renderable element of the component being modified.
+ *
+ * @return string
+ *   The rendered string of the component.
+ *
+ */
+function webform_token_modifier_defaults($modifier, $display_element) {
+  switch ($modifier) {
+    case 'label':
+      return webform_filter_xss($display_element['#title']);
+
+    case 'withlabel':
+      return render($display_element);
+
+    case 'nolabel':
+      $display_element['#theme_wrappers'] = array();
+      return render($display_element);
+
+    case 'key':
+      $values = array();
+      foreach ($display_element['#value'] as $value) {
```

Comment:
We should ensure that `$display_element['#value']` is an array. If for no other reason to get my IDE to stop complaining that it is a string.

```
      if (is_array($display_element['#value'])) {
        foreach ($display_element['#value'] as $value) {
          $values[] = webform_filter_xss($value);
        }
      }
```

Reply:
I've added those, but I'm curious: how did you get the situation where `$display_element['#value']` is a string? In principle, that function is only called for the `key` modifier for select elements, for which it's always an array (or so I'd only ever seen).

Reply:
I tried but couldn't get it to produce a string. But because there are a lot of ifs and foreaches I figure it's best to be explicit on what it can except. My IDE couldn't untangle it either.